### PR TITLE
Add benchmark for reduce_agg

### PR DESCRIPTION
**Pull request by mbasmanova**
Summary:
Add a benchmark to compare performance of reduce_agg vs. sum.

```
============================================================================
[...]l/aggregates/benchmarks/ReduceAgg.cpp     relative  time/iter   iters/s
============================================================================
reduce_agg_global                                            1.11s   904.80m
sum_global                                      66188.%     1.67ms    598.87
reduce_agg_groupby                                           1.14s   873.54m
sum_groupby                                     68175.%     1.68ms    595.54
```

Differential Revision: D49228165

### CODE REVIEW

File: velox/functions/prestosql/aggregates/benchmarks/ReduceAgg.cpp [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+  benchmark->runGroupBySum();
+}
+
+} // namespace
+
+int main(int argc, char** argv) {
+  folly::Init init(&argc, &argv);
+
+  benchmark = std::make_unique<ReduceAggBenchmark>();
+  benchmark->verify();
```
Comment (xiaoxmeng) [inline velox/functions/prestosql/aggregates/benchmarks/ReduceAgg.cpp:228]: This step is to verify benchmark is correct before we actually run the registered benchmarks?

**Review by xiaoxmeng — APPROVED** [review]
@mbasmanova thanks for the benchmark!

File: velox/functions/prestosql/aggregates/benchmarks/ReduceAgg.cpp (hunk 1 above)
Comment (mbasmanova) [reply velox/functions/prestosql/aggregates/benchmarks/ReduceAgg.cpp:228]: Yes. This is to ensure that query plan we compare produce same results.

File: velox/functions/prestosql/aggregates/benchmarks/ReduceAgg.cpp [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    while (auto result = task->next()) {
+      numResultRows += result->size();
+    }
+
+    folly::doNotOptimizeAway(numResultRows);
+  }
+
+  core::PlanFragment makeGroupByPlan(const std::string& aggregate) {
+    return PlanBuilder()
+        .tableScan(inputType_)
```
Comment (kagamiori) [inline velox/functions/prestosql/aggregates/benchmarks/ReduceAgg.cpp:173]: Curious that if the aggregation reads data from table scan, wouldn't this benchmark measure the performance of aggregation in combination with table scan. Is this desired?

Comment (mbasmanova) [reply velox/functions/prestosql/aggregates/benchmarks/ReduceAgg.cpp:173]: Yes, this benchmark measures combined performance sum + table scan vs. reduce_agg + table scan. However, table scan is not a dominant part here. If it was, there would not be 60x difference. Hence, that's Ok.

**Pull request closed without merging**